lib/wiki_api.rb

```diff
@@ -139,4 +142,13 @@ def typical_errors
      Faraday::ConnectionFailed,
      MediawikiApi::HttpError]
   end
+
+  def mediawiki_request_url
+    @mediawiki.instance_variable_get(:@conn).instance_variable_get(:@url_prefix).to_s
+  end
+
+  def save_course_error_record(type_of_error, url)
```

Comment:
It doesn't necessarily need to be done now... the more time you have to think about the ideal way to do it, the better... but it might make sense to extract the error-recording code as a `module` and including it in the API classes that would use it.

Reply:
Okay, I'll shift the code there. Also, it was the same in the three APIs so I anyways had to refactor it somewhere.